packages/sanity/src/core/studio/copyPaste/utils.ts

```diff
@@ -0,0 +1,121 @@
+import {type Path} from '@sanity/types'
+import {isString} from 'sanity'
+
+import {type CopyActionResult} from './types'
+
+const SANITY_CLIPBOARD_ITEM_TYPE = isWebKit()
+  ? 'text/plain'
+  : 'web application/sanity-studio-clipboard-item'
+
+export const getClipboardItem = async (): Promise<CopyActionResult | null> => {
+  const items = await window.navigator.clipboard.read()
+  const item = items.find((i) => i.types.includes(SANITY_CLIPBOARD_ITEM_TYPE))
+  const sanityItem = await item?.getType(SANITY_CLIPBOARD_ITEM_TYPE)
+  if (sanityItem) {
+    const text = await sanityItem.text()
+    return parseCopyResult(text) || null
+  }
+  return null
+}
+
+export const writeClipboardItem = async (copyActionResult: CopyActionResult): Promise<boolean> => {
+  try {
+    const clipboardItem: Record<string, Blob> = {
+      [SANITY_CLIPBOARD_ITEM_TYPE]: new Blob([JSON.stringify(copyActionResult)], {
+        type: SANITY_CLIPBOARD_ITEM_TYPE,
+      }),
+    }
+    if (SANITY_CLIPBOARD_ITEM_TYPE !== 'text/plain') {
+      const text = copyActionResult.items
+        .map((item) => transformValueToText(item.value))
+        .filter(Boolean)
+        .join('\n')
+      if (text.length > 0) {
+        clipboardItem['text/plain'] = new Blob([text], {type: 'text/plain'})
+      }
+    }
+    await window.navigator.clipboard.write([new ClipboardItem(clipboardItem)])
+    return true
+  } catch (error) {
+    console.error(`Failed to write to clipboard: ${error.message}`, error)
+    return false
+  }
+}
+
+export function transformValueToText(value: unknown): string | number {
+  if (!value) {
+    return ''
+  }
+  if (isString(value)) {
+    return value
+  }
+
+  if (Number.isFinite(value)) {
+    return value.toString()
+  }
+
+  if (Array.isArray(value)) {
+    return value
+      .map((item) => transformValueToText(item))
+      .filter(Boolean)
+      .join(', ')
+  }
+
+  if (typeof value === 'object') {
+    const objectValue = value as Record<string, unknown>
+    return Object.keys(objectValue)
+      .map((key) =>
+        key.startsWith('_')
+          ? ''
+          : transformValueToText(typeof value === 'object' ? objectValue[key] : ''),
+      )
+      .filter(Boolean)
+      .join(', ')
+  }
+  return ''
+}
```

Comment:
this function is pretty cool! as far as I understand it takes in a sanity JSON field value and converts it to text for pasting anywhere else? that's awesome

this doesn't have to be a now thing but for later I think we should think about how these values paste into other applications (e.g. google sheets, notion) and optimize its output for that. for example if we joined arrays with new lines instead of commas, it would paste into google sheets as different rows

cc @runeb for future thoughts (potentially outside of the scope of this PR tbh)